### PR TITLE
allow tasty version to be inspected

### PR DESCRIPTION
**Pull request by bishabosha**
this is intended to allow the tasty version to be extracted, should we need to eventually branch code based on the version

### CODE REVIEW

File: tasty/src-bootstrapped/dotty/tools/tasty/TastyHeaderUnpickler.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    val h = readFullHeader()
+    val validVersion = (
+      h.majorVersion == MajorVersion
+      && h.minorVersion >= 0 && h.minorVersion <= MinorVersion
+    )
+    check(validVersion,
+      s"""TASTy signature has wrong version.
+         | expected: $MajorVersion.$MinorVersion
+         | found   : ${h.majorVersion}.${h.minorVersion}""".stripMargin
+    )
```
Comment (sjrd) [inline tasty/src-bootstrapped/dotty/tools/tasty/TastyHeaderUnpickler.scala:26]: Is there any reason not to do this in `readFullHeader()`?

Comment (bishabosha) [reply tasty/src-bootstrapped/dotty/tools/tasty/TastyHeaderUnpickler.scala:26]: I made an iteration where the user passed in a custom range to validate against, but then perhaps they may want to apply custom logic not anticipated

Comment (sjrd) [reply tasty/src-bootstrapped/dotty/tools/tasty/TastyHeaderUnpickler.scala:26]: > but then perhaps they may want to apply custom logic not anticipated

Don't prepare for completely unknown use cases. The use cases we know about will always want the validation, so we should validate. In the unlikely event that someone wants a non-validating variant, we can add another method in the future whose name clearly says that it doesn't validate.

Comment (bishabosha) [reply tasty/src-bootstrapped/dotty/tools/tasty/TastyHeaderUnpickler.scala:26]: should the version to validate against be passed as a parameter?

Comment (sjrd) [reply tasty/src-bootstrapped/dotty/tools/tasty/TastyHeaderUnpickler.scala:26]: I don't think so. Simply do as `readHeader()` currently does.

**Comment by bishabosha** [comment]
looks like the bootstrapping of sources doesn't work as I expected

File: tasty/src-bootstrapped/dotty/tools/tasty/TastyHeaderUnpickler.scala (hunk 1 above)
Comment (bishabosha) [reply tasty/src-bootstrapped/dotty/tools/tasty/TastyHeaderUnpickler.scala]: done

**Review by sjrd — APPROVED** [review]

**Pull request merged**